### PR TITLE
feat: Refactor lottery contract and test to allow 4 types of nfts wit…

### PR DESCRIPTION
**Pull request by pablin-dev**
feat: Refactor lottery contract and test to allow 4 types of nfts with different probabilities to be used in a modular way.

I will add soon the UML architecture diagram, but code is ready to review.

### CODE REVIEW

File: contracts/apps/nft.sol [hunk 1]
```diff
@@ -2,50 +2,77 @@
...
+     * @dev Mints `quantity` tokens and transfers them to `owner`.
+     *
+     * Requirements:
+     *
+     * - `to` cannot be the zero address.
+     * - `quantity` must be greater than 0.
+     *
+     * Emits a {Transfer} event for each mint.
+     */
+    function mint(uint256 quantity) external onlyOwner {
```
Comment (coderSomya) [inline contracts/apps/nft.sol:35]: i think we should also take the metadata of the NFT to be minted here.

Comment (pablin-dev) [reply contracts/apps/nft.sol]: Changed ERC721A to ERC1155 as suggested on the discord.

File: contracts/apps/lazy1155.sol [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    function pause() public onlyOwner {
+        _pause();
+    }
+
+    function unpause() public onlyOwner {
+        _unpause();
+    }
+
+    function mint(address account, uint256 id, uint256 amount, bytes memory data) public onlyOwner {
```
Comment (coderSomya) [inline contracts/apps/lazy1155.sol:37]: can we keep a check in ```mint``` and ```mintBatch``` to first check if the tokenCap isn't exceeded? something like
```
if (id >= _tokenCap) {
    revert TooManyTokensMinted();
}
 ```

Comment (pablin-dev) [reply contracts/apps/lazy1155.sol:40]: Good point. Added.

**Comment by pablin-dev** [comment]
Its break when merging manually, better create a new `clean` PR.

**Pull request closed without merging**